src/client/remote_nodestream.ml

```diff
@@ -83,12 +83,19 @@ class remote_nodestream ((ic,oc) as conn) =
             end
         end
       in
+      let _should_reinit = ref false in
       let rec loop_parts () =
         Llio.input_int ic >>= function
         | (-2) -> Logger.info_f_ "loop_parts done"
         | 1 ->
           begin
             Logger.debug_f_ "loop_entries" >>= fun () ->
+            begin
+              if !_should_reinit
+              then tlog_coll # reinit()
```

Comment:
the `new_entry` function keeps track of when a roll over is needed, but it doesn't know that whole files were injected meanwhile. It's indeed possible to push this down and make it part of the state management as a side effect of `save_tlog_file`. I'll do that.